src/ui-test/camelk_java_debug_test.ts

```diff
@@ -0,0 +1,245 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License", destination); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+'use strict'
+
+import { assert } from 'chai';
+import path = require('path');
+import * as pjson from '../../package.json';
+import {
+	ActivityBar,
+	CustomTreeSection,
+	DebugToolbar,
+	EditorView,
+	InputBox,
+	SideBarView,
+	TextEditor,
+	ViewItem,
+	VSBrowser,
+	Workbench
+} from 'vscode-extension-tester';
+import { Marketplace } from 'vscode-uitests-tooling';
+import { prepareEmptyTestFolder } from './utils/resourcesUtils';
+import { extensionIsActivated, sectionHasItem } from './utils/waitConditions';
+import * as uiTestConstants from './utils/uiTestConstants';
+
+const TEST_FOLDER = '../../../testFolder';
+const WORKSPACE_FOLDER = path.join(__dirname, TEST_FOLDER);
+
+const START_DEBUG_LABEL = uiTestConstants.startDebug;
+const REMOVE_INTEGRATION_LABEL = uiTestConstants.integrationRemove;
+
+describe('Test Debug on Camel K Integrations from Side Bar', function () {
+
+	before(async function () {
+		this.timeout(90000);
+		this.retries(3);
+		await VSBrowser.instance.driver.sleep(20000);
+		await VSBrowser.instance.waitForWorkbench;
+		await prepareTempWorkspaceForTests(WORKSPACE_FOLDER);
+		await VSBrowser.instance.driver.wait(camelKToolingIsEnabled);
+	});
+
+	describe('Java Debug', function () {
+		
+		const INTEGRATION_LABEL = 'java-debug-test';
+		const INTEGRATION_FILE = 'JavaDebugTest';
+
+		before(async function (){
+			this.timeout(20000);
+			await moveToExplorerActivity();
+			await createIntegration(INTEGRATION_FILE);
+			await startIntegration(INTEGRATION_LABEL);
+		})
+
+		it('Check Java Debug available', async function () {
+			this.timeout(20000);
+
+			const item = await getIntegration(INTEGRATION_LABEL);
+			const menu = await item.openContextMenu();
+
+			assert.isTrue(await menu.hasItem(START_DEBUG_LABEL));
```

Comment:
This needs a conditional waiting because the context menu becomes available only when the integration is in state "running". Which can take several seconds/minute it this test is played first

Reply:
I think this must be adressed

Reply:
Sorry, I didn't address this.

This is not needed because StartIntegration doesn't resolve until the integration is created/visible. Check:https://github.com/camel-tooling/vscode-camelk/pull/1190/files/8993fea59594abb190d6e85a7fdaaf4af18fee97#diff-da62a10e4200983f607ee8aeecd952e9941cdc88b1bc1c9ab1a93860ef672efbR198

Reply:
I think it is waiting that the integration is visible but not that it is in "running" state. On first run, the "building" phase can take several minutes.

Reply:
Oh, I see now. Ok, I'll put it